### PR TITLE
Bump to 0.10.0

### PR DESCRIPTION
**Pull request by CedNaru**
Bump version to be ready for the next release.

I also commit minor changes that don't deserve their own PR. They are meant to solve issues discovered while updating the 3D demo.

- Window - Editor, JVM export, Native Image export tested
- Android - Export tested

### CODE REVIEW

File: docs/src/doc/user-guide/advanced/graal-vm-native-image.md [hunk 1]
```diff
@@ -43,9 +42,9 @@ The same applies for resource files which should be added (basically any files i
 ```kotlin
 godot {
     isGraalExportEnabled.set(true)
```
Comment (chippmann) [inline docs/src/doc/user-guide/advanced/graal-vm-native-image.md:44]: this one needs to change as well no?

**Comment by chippmann** [comment]
Linux editor and exports work on the jvm
will test graalvm later

**Comment by chippmann** [comment]
Tested on linux with graalvm as well.
Everything works except the jvm does not close.

**Comment by CedNaru** [comment]
Confirming the issue with native image.
It seems the daemon thread trick doesn't work with native images.

**Comment by chippmann** [comment]
Works now for graalvm as well

**Review by chippmann — DISMISSED** [review]

**Review by chippmann — DISMISSED** [review]

File: kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/projectExt/setupConfigurationsAndCompilations.kt [hunk 2]
```diff
@@ -33,6 +33,7 @@ fun Project.setupConfigurationsAndCompilations() {
         with(it.dependencies) {
             add(dependencies.create("org.jetbrains.kotlin:kotlin-stdlib:${kotlinJvmExtension.coreLibrariesVersion}"))
             add(dependencies.create("com.utopia-rise:$godotLibraryArtifactName:${GodotBuildProperties.assembledGodotKotlinJvmVersion}"))
+            add(dependencies.create("com.utopia-rise:tools-common:${GodotBuildProperties.assembledGodotKotlinJvmVersion}"))
```
Comment (piiertho) [inline kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/projectExt/setupConfigurationsAndCompilations.kt:36]: Why do we add this one ?

File: src/script/jvm_script_manager.h [hunk 3]
```diff
@@ -25,56 +29,54 @@ class JvmScriptManager {
...
     template<class SCRIPT>
-    Ref<SCRIPT> get_or_create_script(const String& p_path);
-
+    Ref<SCRIPT> get_or_create_script(const String& p_path, bool* created);
     void clear();
+
+#ifdef TOOLS_ENABLED
+    int64_t get_last_reload();
+
+    void invalidate_source(String path);
```
Comment (piiertho) [inline src/script/jvm_script_manager.h:48]: Why not a const ref ?

File: kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/projectExt/setupConfigurationsAndCompilations.kt (hunk 2 above)
Comment (CedNaru) [reply kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/projectExt/setupConfigurationsAndCompilations.kt:36]: I need tools-common in the bootstrap, it has utilities required at runtime.
It also fixes an inconsistency. The boostrap generated for the editor had it, but not the one generated by the gradle build.

Comment (piiertho) [reply kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/projectExt/setupConfigurationsAndCompilations.kt:36]: We need to fix this after next release as we add kotlin poet to user classpath with this one.

**Review by piiertho — APPROVED** [review]

**Pull request merged**